core/app/models/spree/product_property.rb

```diff
@@ -7,7 +7,8 @@ class ProductProperty < Spree::Base
       belongs_to :property, class_name: 'Spree::Property'
     end
 
-    validates :property, presence: true
+    validates :product, presence: true
+    validates :property, uniqueness: { scope: :product }, presence: true
```

Comment:
@dilpreet92 We should add `message: :already_linked` with uniqueness validation.